src/mode.ts

```diff
@@ -53,11 +55,11 @@ export function getTypeScriptWorker(): Promise<TypeScriptWorker> {
 	});
 }
 
-function setupMode(defaults:LanguageServiceDefaultsImpl, modeId:string, language:Language): (first: Uri, ...more: Uri[]) => Promise<TypeScriptWorker> {
+function setupMode(defaults:LanguageServiceDefaultsImpl, modeId:string, language:Language, stopWorkerWhenIdleFor?: number): (first: Uri, ...more: Uri[]) => Promise<TypeScriptWorker> {

```

Comment:
Instead of adding another parameter to `setupMode` I think it's better to expose this via the `LanguageServiceDefaults` interface. That's already exported as API and (also) allows per language configuration.